pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -118,6 +120,30 @@ func (n *nsLifecycle) ensurePRTBAddToNamespace(ns *v1.Namespace) (bool, error) {
 			return false, errors.Wrapf(err, "couldn't ensure binding %v in %v", prtb.Name, ns.Name)
 		}
 	}
+
+	prtbSet := map[string]string{}
+	prtbList := n.m.prtbIndexer.List()
+	for _, prtb := range prtbList {
+		prtb, ok := prtb.(*v3.ProjectRoleTemplateBinding)
+		if !ok {
+			return false, err
+		}
+		prtbSet[convert.ToString(prtb.GetUID())] = prtb.Namespace
+	}
+	rbs, _ := n.m.workload.K8sClient.RbacV1().RoleBindings(ns.Name).List(metav1.ListOptions{})
+	client := n.m.workload.RBAC.RoleBindings(ns.Name).ObjectClient()
+	for _, rb := range rbs.Items {
+		uid := convert.ToString(rb.Labels[rtbOwnerAnn])
+		if pns, ok := prtbSet[uid]; ok {
+			if pns != projectID {
+				logrus.Debugf("Deleting Role Binding %s", rb.Name)
+				if err := client.Delete(rb.Name, &metav1.DeleteOptions{}); err != nil {
+					return false, err
+				}
+			}
+		}
+	}
+
```

Comment:
Logic seems overly complex and has the flaw of preloading all prtbs in memory.
I think the logic should be like this:
```
rbs := n.m.rbLister.List(...) // get all RBs in the namespace, from the cache
for _, r := range rbs {
   if ownerUID, ok := r.Labels[rtbOwnderLabel]; ok { // do the ok check so that we only delete rbs that we are managing
     prtb, err := n.m.prtbIndexer(prtbByUIDIndex, ownerUID)
     if prtb.Namespace != namespacesCurrentProject {
        delete
     }
   }
}
```